daml-assistant/src/DAML/Assistant/Install/Github.hs

```diff
@@ -83,41 +42,42 @@ tagToVersion (Tag t) =
         else
             Left "Tag must start with v followed by semantic version."
 
--- | Make a request to the Github API. There is an unauthenticated user rate limit
--- of 60 requests per hour, so we should limit requests. (One way for the user to
--- avoid API requests is to give the version directly.)
-makeAPIRequest :: FromJSON t => Text -> IO t
-makeAPIRequest path = do
-    request <- parseRequest (unpack ("GET https://api.github.com/repos/digital-asset/daml" <> path))
-    response <- httpJSON (setRequestHeader "User-Agent" ["daml"] request)
-    when (getResponseStatusCode response /= 200) $
-        throwString . show $ getResponseStatus response
-    pure (getResponseBody response)
+-- | Get the version of the latest stable (i.e. non-prerelease) release.
+-- We avoid the Github API because of very low rate limits. As such, we
+-- discover the latest version by parsing an HTTP redirect. We make a
+-- request to:
+--
+--     https://github.com/digital-asset/daml/releases/latest
+--
+-- Which always redirects to the latest stable release, for example:
+--
+--     https://github.com/digital-asset/daml/releases/tag/v0.12.3
+--
+-- So we take that URL to get the tag, and from there the version of
+-- the latest stable release.
+getLatestVersion :: IO SdkVersion
+getLatestVersion = do
+    request <- parseRequest "GET https://github.com/digital-asset/daml/releases/latest"
+    response <- httpNoBody request { redirectCount = 0 }
+
+    when (getResponseStatusCode response /= 302) $
+        failed "Expected response status 302."  response
```

Comment:
Can we make this message a bit clearer, it would confuse the hell out of me if I got it. Expected url <insert url here> to return response code 302, but returned something else. Or similar.